CRM/Core/BAO/Setting.php

```diff
@@ -519,17 +513,13 @@ public static function isAPIJobAllowedToRun($params): void {
    */
   public static function onChangeEnvironmentSetting($oldValue, $newValue, $metadata) {
     if ($newValue !== 'Production') {
-      $mailing = Civi::settings()->get('mailing_backend');
-      if ($mailing['outBound_option'] != CRM_Mailing_Config::OUTBOUND_OPTION_DISABLED) {
-        Civi::settings()->set('mailing_backend_store', $mailing);
-      }
-      Civi::settings()->set('mailing_backend', ['outBound_option' => CRM_Mailing_Config::OUTBOUND_OPTION_DISABLED]);
       CRM_Core_Session::setStatus(ts('Outbound emails have been disabled. Scheduled jobs will not run unless runInNonProductionEnvironment=TRUE is added as a parameter for a specific job'), ts("Non-production environment set"), "success");
+      Civi::settings()->set('mailing_backend', ['outBound_option' => CRM_Mailing_Config::OUTBOUND_OPTION_DISABLED]);
     }
-    else {
-      $mailing = Civi::settings()->get('mailing_backend_store');
-      if ($mailing) {
-        Civi::settings()->set('mailing_backend', $mailing);
+    if ($newValue == 'Production' && $oldValue != 'Production') {
+      $mailing_backend = Civi::settings()->set('mailing_backend');
```

Comment:
There's two problems causing test fails. The simple one is that this seems intended to be `get` not `set`.